docs/source/trait_types.rst

```diff
@@ -0,0 +1,108 @@
+Trait Types
+===========
+
+.. module:: traitlets
+
+.. class:: TraitType
+
+   The base class for all trait types.
+
+   .. automethod:: __init__
+
+Numbers
+-------
+.. autoclass:: Int
+
+.. class:: Long
+
+   Integers ``> sys.maxsize`` on Python 2. Alias for :class:`Int` on Python 3.
+
+.. class:: Integer
```

Comment:
Can there be a note on Int and Long that under ~all circumstances, Integer should be used instead? Int and Long should only be used when the opposite would be invalid, which is quite rare.